File: concept/14_string/reverse_word/reverse_word.dart
void main() {
  // Reverse Word
  // ym eman si nitin
  String s = "my name is test nitin gamechi";
  int start = 0;

  String rSubString = '';
  int size = s.length;
  s = s + " ";
  for (int i = 1; i <= size; i++) {
    if (s[i] == ' ' || i == size) {
      int end = i;

      String substring = s.substring(start, end);
      String reversedWord = substring.split('').reversed.join();
      rSubString += reversedWord + " ";
      start = i + 1;
    }
    print(rSubString);
  }

  print(rSubString);
}

File: concept/14_string/reverse_word/reverse_word.cpp
#include <iostream>
using namespace std;

int main()
{
    // Reverse Word
    // ym eman si nitin
    string s = "my name is test nitin gamechi";
    int start = 0;

    for (int i = 0; i <= s.length(); i++)
    {

        if (s[i] == ' ' || s[i] == '\0' || i == s.length())
        {
            int end = i - 1;

            while (start <= end)
            {
                swap(s[start++], s[end--]);
            }

            start = i + 1;
        }
    }

    cout << s << endl;
}